src/pytorch_lightning/tuner/lr_finder.py

```diff
@@ -40,7 +40,7 @@
 _MATPLOTLIB_AVAILABLE = RequirementCache("matplotlib")
 if _MATPLOTLIB_AVAILABLE and TYPE_CHECKING:
```

Comment:
Using this PR for this fix

```suggestion
if TYPE_CHECKING and _MATPLOTLIB_AVAILABLE:
```